kong/utils.go

```diff
@@ -741,3 +741,119 @@ func FillPluginsDefaults(plugin *Plugin, schema Schema) error {
 func FillPluginsDefaultsWithOpts(plugin *Plugin, schema map[string]interface{}, opts FillRecordOptions) error {
 	return fillConfigRecordDefaultsAutoFields(plugin, schema, opts)
 }
+
+func deleteAndCollapseMap(config map[string]interface{}, path []string) {
+	key := path[0]
+	if len(path) == 1 {
+		delete(config, key)
```

Comment:
Hmm... does it matter 🤔 ? I mean - it won't enter `else` block and having a `return` right after this line doesn't change anything right? Like there won't be any other instruction either way.


Oh... maybe you mean like:
```go
if condition {
  return do_something()
}
// here would be the else part?
```

?
I'm not sure I fully understand 🤔

Reply:
I meant something like this:

```
  if condition {
    // do something
    return
  }
  
  // rest of the code, no else
```

This helps with readability as well.

Reply:
Oh... Ok. Honestly I think this is just a matter of taste. For me the current implementation reads better. Is it a _"go" thing_ to add returns at the end of `if / else` blocks? I can switch it if that's how it's usually done in go, otherwise I'd prefer to leave it as it is.

Reply:
+1 to what @Prashansa-K suggests. This way you keep the happy path the least indented one, plus be done with the if statement: there might be other conditions which you'll need to handle below).

Reply:
Alright, I've changed the code to requested style :)